app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java

```diff
@@ -1932,18 +1906,18 @@ public void onReceive(Context context, Intent intent) {
     }
 
     /**
-     * Request edit of image (by image editor/viewer - for example to crop image)
+     * Request edit of file
      *
      * @param file File that should be edited
      */
-    public void requestPictureEdit(final Context context, final File file) {
-        final Uri uri = getUriByFileProviderAuthority(context, file);
+    public void requestFileEdit(final Context context, final File file) {
+        final Uri uri = FileProvider.getUriForFile(context, getFileProvider(context), file);
         final Intent intent = new Intent(Intent.ACTION_EDIT);
-        intent.setDataAndType(uri, "image/*");
-        intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION | Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        intent.setData(uri);  // Mime type inferred automatically
```

Comment:
Sure that this always works as expected? While the original intent here was to edit pictures and thus image/ was specified, I think it makes sense to keep it this way to explicitly give additional info.

My idea is to use the Fileutils mimetype method here and keep using setDataAndType

Reply:
It seems to work both ways. Using your approach now.